sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateFormatter.scala

```diff
@@ -93,7 +93,8 @@ class LegacySimpleDateFormatter(pattern: String, locale: Locale) extends LegacyD
 object DateFormatter {
   import LegacyDateFormats._
 
-  val defaultLocale: Locale = Locale.US
+  // Default locale should be obtained from JVM defaults.
+  val defaultLocale: Locale = Locale.getDefault
```

Comment:
@HyukjinKwon @MaxGekk , default locale should be obtained from local JVM settings, is this right?

Reply:
Changing the default locale is slightly risky because it is involved in resolving of short time zone names inside of DateTimeFormatter.

`Locale.US` is used in Spark SQL everywhere as default locale. I am wondering why do you need to change it?

Reply:
Locales affects week based dates. But since US is consider as a default, users should know that they have to specify that while computing timestamps or dates.